lms/djangoapps/course_api/blocks/transformers/tests/test_milestones.py

```diff
@@ -166,7 +166,7 @@ def test_gated(self, gated_block_ref, gating_block_ref, expected_blocks_before_c
         self.course.enable_subsection_gating = True
         self.setup_gated_section(self.blocks[gated_block_ref], self.blocks[gating_block_ref])
 
-        with self.assertNumQueries(5):
+        with self.assertNumQueries(14):
```

Comment:
Question: Looking at the queries files that you generated (which are awesome, BTW), my understanding is that this increase is a result of building the block structures entry for the course, and all the corresponding transaction save points. Does this go down to 6 queries if you run the block structure collection/caching before this assert queries block?

It might be worth noting in a query that we don't ever expect it to actually do these queries in a production setting, since these records would have been written on-publish.